### DISCUSSION TITLE
Dropping in-flight connect requests

### DISCUSSION THREAD

**Discussion by anilaltuner** [libp2p/rust-libp2p#5558] [General]
# Libp2p-relay Capacity Warning

## Issue Description

I've been working with libp2p relays and these following warnings in my logs:

```
[2024-08-14T12:16:06Z WARN libp2p_relay::priv_client::handler] Dropping in-flight connect request because we are at capacity
```

This warning seems to indicate that the relay client is dropping connection requests due to reaching some kind of capacity limit. 

## Questions

1. What exactly does "at capacity" mean in this context? Is there a configurable limit for the number of in-flight connect requests?

2. Could this warning indicate a potential bottleneck in our network topology or relay configuration?

**Comment by dariusc93** [comment]
This seems to happen due to the concurrent streams for that connection is hitting its max

https://github.com/libp2p/rust-libp2p/blob/4d7a535c7fa750ba88622e7d7cbbbbbb6884985c/protocols/relay/src/priv_client/handler.rs#L51

The error would happen, as you noticed, during an establishment of a circuit
https://github.com/libp2p/rust-libp2p/blob/4d7a535c7fa750ba88622e7d7cbbbbbb6884985c/protocols/relay/src/priv_client/handler.rs#L200-L227

I think ive only ever hit that error once but cant recall my solution, however if youre establishing multiple circuit to the peer via that relay, that *may* contribute to that error. Do you have any code you could share to show how youre handling connections or if you have any sample code that could reproduce that error?
